### PR TITLE
add create_unicef_user pipeline step

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

**Review by reinbach — APPROVED** [review]

**Comment by robertavram** [comment]
can we add a test to this

**Comment by domdinicola** [comment]
@robertavram added!

File: src/unicef_security/pipeline.py [hunk 1]
```diff
@@ -17,3 +18,19 @@ def social_details(backend, details, response, *args, **kwargs):
...
+
+
+def create_unicef_user(strategy, details, backend, user=None, *args, **kwargs):
+    """Overrides create_user, to create only UNICEF users"""
+    if user:
+        return {'is_new': False}
+
+    fields = dict((name, kwargs.get(name, details.get(name)))
+                  for name in backend.setting('USER_FIELDS', USER_FIELDS))
+    if not (fields and details.get('email', '').endswith('@unicef.org')):
```
Comment (robertavram) [inline src/unicef_security/pipeline.py:30]: can we move the `endswith` string ("@unicef.org") into a setting?

**Review by robertavram — CHANGES_REQUESTED** [review]

File: src/unicef_security/pipeline.py (hunk 1 above)
Comment (domdinicola) [reply src/unicef_security/pipeline.py]: Done!
Do you see "@unicef.org" to change in the future?

Comment (reinbach) [reply src/unicef_security/pipeline.py]: @domdinicola It allows one to easily override the setting in tests, so you're not required to use `@unicef.org` in the tests and can instead make use of `@example.com`

File: tests/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        'iss': 'https://tenant.b2clogin.com/1234567890/v2.0/',
+        'sub': 'abcdefgh', 'aud': 'abcdefgh',
+        'acr': 'b2c_1a_unicef_social_signup_signin',
+        'iat': 1607541825,
+        'auth_time': 1607541824,
+        'given_name': 'Given',
+        'family_name': 'Family',
+        'name': 'Given Family',
+        'idp': 'UNICEF Azure AD',
+        'email': 'test_undeliverable@unicef.org'
```
Comment (reinbach) [inline tests/test_pipeline.py:40]: I'd suggest making use of `@example.com` in tests

**Review by reinbach — APPROVED** [review]

File: tests/test_pipeline.py (hunk 2 above)
Comment (domdinicola) [reply tests/test_pipeline.py:40]: and override the env var?

Comment (reinbach) [reply tests/test_pipeline.py:40]: You can pass in the `settings` param in the test and then override that specific setting e.g.
See https://pytest-django.readthedocs.io/en/latest/helpers.html#settings

Comment (robertavram) [reply tests/test_pipeline.py:40]: On the next todo: use the config file to try and get the settings from the project settings file and set defaults. 
As part of that work we can override the UNICEF address in the tests.

**Review by robertavram — APPROVED** [review]

**Pull request merged**